### PR TITLE
prognostic run: save plots to tensorboard

### PR DESCRIPTION
**Pull request by nbren12**
Tensorboard is a great way to monitor a long-running computation like an FV3
run with online learning. This PR adds an simple switch to allow visualizing 2D
fields in tensorboard.

Added public API:
- new configuration option `user_config.diagnostics[].tensorboard`

- [x] Tests added

### CODE REVIEW

File: workflows/prognostic_c48_run/runtime/diagnostics/tensorboard.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+def plot_to_image(figure):
+    """Converts the matplotlib plot specified by 'figure' to a PNG image and
+    returns it. The supplied figure is closed and inaccessible after this call."""
+    # Save the plot to a PNG in memory.
+    buf = io.BytesIO()
+    plt.savefig(buf, format="png")
+    # Closing the figure prevents it from being displayed directly inside
+    # the notebook.
+    plt.close(figure)
```
Comment (frodre) [inline workflows/prognostic_c48_run/runtime/diagnostics/tensorboard.py:21]: Why does the close appear here and not in the context where the figure is created?  Maybe adjusting `savefig` above means it doesn't have to be closed here?

**Review by frodre — APPROVED** [review]
Thanks, @nbren12 for the update and showing me what it looks like in action.  Minor comments relating to the figure saving/closing, but overall looks good!

File: workflows/prognostic_c48_run/sklearn_runfile.py [hunk 2]
```diff
@@ -38,23 +38,26 @@
...
+                comm, diagnostics, ["specific_humidity_limiter_active"]
+            )
+            if comm.rank == 0:
+                runtime.log_mapping(time, averages, "statistics")
+                runtime.log_mapping(time, profiles, "profiles")
 
-        for diag_file in diag_files:
-            diag_file.observe(time, diagnostics)
+            for diag_file in diag_files:
+                diag_file.observe(time, diagnostics)
```
[the remark this answers is not in the document]
Comment (nbren12) [reply workflows/prognostic_c48_run/sklearn_runfile.py:60]: That's correct. The variable and averaging/time selection is handled by `DiagnosticFile`, which could probably be renamed now that it works with different sinks. Keeping with the plumbing metaphor, it's more of a "faucet" now.

File: workflows/prognostic_c48_run/runtime/diagnostics/tensorboard.py (hunk 1 above)
Comment (nbren12) [reply workflows/prognostic_c48_run/runtime/diagnostics/tensorboard.py:21]: I'll remove it.

**Pull request merged**